tools/bigquery-zos-mainframe-connector/src/main/scala/com/google/cloud/bqsh/op/MergeInto.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Google LLC All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bqsh.op
+
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.cloud.bigquery._
+import com.google.cloud.bqsh.BQ
+import com.google.cloud.gszutil.Util
+import com.google.cloud.gszutil.Util.Logging
+
+object MergeInto extends Logging {
+  def run(bq: BigQuery, source: TableId, target: TableId, naturalKeyColumns: Seq[String], creds: GoogleCredentials, dryRun: Boolean): Unit = {
+    val mergeRequest = buildMergeRequest(bq, source, target, naturalKeyColumns)
+
+    val query = genMerge(mergeRequest)
+    logger.debug("Generated SQL:\n" + query)
+
+    val cfg = QueryJobConfiguration.newBuilder(query)
+      .setUseLegacySql(false)
+      .setAllowLargeResults(true)
+      .setUseQueryCache(false)
+      .setPriority(QueryJobConfiguration.Priority.BATCH)
+      .setDryRun(dryRun)
+      .build()
+
+    val jobId: JobId = JobId.of(s"gszutil_merge_into_${target.getDataset}_${target.getTable}_${System.currentTimeMillis()/1000}_${Util.randString(6)}")
+    BQ.runJob(bq, cfg, jobId, 60*60)
+  }
+
+  case class MergeRequest(a: TableId, b: TableId, naturalKeyCols: Seq[String], valCols: Seq[String])
+
+  def getSchema(t: Table): Schema = {
+    if (t.getDefinition == null) {
+      t.reload()
+    }
+    t.getDefinition[TableDefinition].getSchema
+  }
+
+  def buildMergeRequest(bq: BigQuery, source: TableId, target: TableId, naturalKeyCols: Seq[String]): MergeRequest = {
+    val tblA = bq.getTable(target)
+    val tblB = bq.getTable(source)
+    val schemaA = getSchema(tblA)
+    val schemaB = getSchema(tblB)
+    val fields = getFields(schemaA)
+    val naturalKeySet = naturalKeyCols.toSet
+    validateMerge(fields.toMap, getFields(schemaB).toMap, naturalKeySet)
+    val valCols = fields.map(_._1).filterNot(naturalKeySet.contains)
```

Comment:
I don't think the set difference operator is easily recognized and also conversion to set will lose the ordering.
Either way I'm removing this entire class because it's not actually used.